bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -91,27 +93,25 @@ public ForecastWeatherHandler(Thing thing) {
 
     @Override
     public void initialize() {
-        try {
-            Object location = getConfig().get(BindingConstants.LOCATION);
-            if (location == null) {
-                logger.debug("Location not set for thing {} -- aborting initialization.", getThing().getUID());
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        String.format("location parameter not set"));
-                return;
-            }
-            String latlon = location.toString();
-            String[] split = latlon.split(",");
-            if (split.length != 2) {
-                throw new NumberFormatException(String.format(
-                        "Expecting location parameter to have latitude and longitude separated by comma (LATITUDE,LONGITUDE). Found %d values instead.",
-                        split.length));
-            }
-            this.location = new LatLon(new BigDecimal(split[0].trim()), new BigDecimal(split[1].trim()));
-            super.initialize();
-        } catch (NumberFormatException e) {
+        ForecastConfiguration config = getConfigAs(ForecastConfiguration.class);
+        String location = config.location;
+        if (location.isBlank()) {
+            logger.debug("Location not set for thing {} -- aborting initialization.", getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter not set"));
```

Comment:
No need for String.format ... as there is nothing to format ;)

Reply:
Thanks, I didn't notice when simplifying the `initialize` method. 🙂 Fixed, and also removed redundant logging.